From 7e4a164c3491fa5e38014d4876dbd5a95fd25b91 Mon Sep 17 00:00:00 2001
From: Frank Hunleth <fhunleth@troodon-software.com>
Date: Sat, 18 Feb 2017 15:53:44 -0500
Subject: [PATCH] erlang: set ERL_TOP to avoid using user's version

Explicitly set the ERL_TOP environment variable to override values set
on a user's system. This prevents Erlang's configure scripts from
using source files outside of the ones that Buildroot downloaded.

Signed-off-by: Frank Hunleth <fhunleth@troodon-software.com>
---
 package/erlang/erlang.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/package/erlang/erlang.mk b/package/erlang/erlang.mk
index 66e9842..136f66b 100644
--- a/package/erlang/erlang.mk
+++ b/package/erlang/erlang.mk
@@ -36,6 +36,12 @@ ERLANG_CONF_ENV += erl_xcomp_clock_gettime_cpu_time=yes
 
 ERLANG_CONF_OPTS = --without-javac
 
+# Force ERL_TOP to the downloaded source directory. This prevents
+# Erlang's configure script from inadvertantly using files from
+# a version of Erlang installed on the host.
+ERLANG_CONF_ENV += ERL_TOP=$(@D)
+HOST_ERLANG_CONF_ENV += ERL_TOP=$(@D)
+
 # erlang uses openssl for all things crypto. Since the host tools (such as
 # rebar) uses crypto, we need to build host-erlang with support for openssl.
 HOST_ERLANG_DEPENDENCIES = host-openssl
-- 
2.7.4

